### PR TITLE
docs: edit model debug doc

### PR DESCRIPTION
**Pull request by dzhu**
## Description

- Use the signup link for all links to the community Slack, since that
  redirects to the actual workspace if the user is already signed
  up (and perhaps the links will mostly be clicked by people who haven't
  signed up anyway).

- Remove indentation that led to unnecessary blockquote elements.

- Make other miscellaneous copy edits.

## Test Plan

- [x] render and examine docs

### CODE REVIEW

**Review by rb-determined-ai — APPROVED** [review]
Thank you!

File: docs/how-to/model-debug.txt [hunk 1]
```diff
@@ -17,69 +17,69 @@ is assumed that you have already successfully :ref:`installed Determined
...
 
 The basic steps for debugging are:
 
 **Model-related issues:**
 
-   #. Does the original code run locally?
-   #. Does each method of your Trial class work locally?
-   #. Does local test mode work?
+#. Does the original code run locally?
+#. Does each method of your trial class work locally?
```
[the remark this answers is not in the document]
Comment (dzhu) [reply docs/how-to/model-debug.txt:33]: Hmm, in that case it should probably be more like "`Trial` subclass", plus I find it a little strange that that sense of `Trial` is not actually introduced here. I can massage things to use that some of the time.

[the remark this answers is not in the document]
Comment (rb-determined-ai) [reply docs/how-to/model-debug.txt]: This isn't a "you're new to determined" document, it's a "you've seen the api reference docs but you don't know how to make your Trial class actually work" kind of document, so I'm fine with that.  Maybe mentioning once that your Trial is one of PyTorchTrial, KerasTrial, or EstimatorTrial would suffice?

[the remark this answers is not in the document]
Comment (dzhu) [reply docs/how-to/model-debug.txt]: Cool, that's about where I ended up.

**Pull request merged**